sqlmesh/core/model/definition.py

```diff
@@ -761,6 +769,10 @@ def seed_path(self) -> Path:
             return self._path.parent / seed_path
         return seed_path
 
+    @property
+    def has_self_reference(self) -> bool:
```

Comment:
Do we want the same override for `ExternalModel`?

Reply:
It is not needed. Seed needed it because of an error I got when it does a columns_to_type thing.